_search-plugins/concurrent-segment-search.md

```diff
@@ -117,8 +196,18 @@ Non-concurrent search calculates the document count error and returns it in the
 For more information about how `shard_size` can affect both `doc_count_error_upper_bound` and collected buckets, see [this GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/11680#issuecomment-1885882985).
 
 
-## Developer information: AggregatorFactory changes
+## Developer information: 
+
+### AggregatorFactory changes
 
 Because of implementation details, not all aggregator types can support concurrent segment search. To accommodate this, we have introduced a [`supportsConcurrentSegmentSearch()`](https://github.com/opensearch-project/OpenSearch/blob/bb38ed4836496ac70258c2472668325a012ea3ed/server/src/main/java/org/opensearch/search/aggregations/AggregatorFactory.java#L121) method in the `AggregatorFactory` class to indicate whether a given aggregation type supports concurrent segment search. By default, this method returns `false`. Any aggregator that needs to support concurrent segment search must override this method in its own factory implementation. 
 
 To ensure that a custom plugin-based `Aggregator` implementation works with the concurrent search path, plugin developers can verify their implementation with concurrent search enabled and then update the plugin to override the [`supportsConcurrentSegmentSearch()`](https://github.com/opensearch-project/OpenSearch/blob/bb38ed4836496ac70258c2472668325a012ea3ed/server/src/main/java/org/opensearch/search/aggregations/AggregatorFactory.java#L121) method to return `true`.
+
+### Pluggable concurrent search deciders: ConcurrentSearchRequestDecider
+
+Plugin developers can customize the concurrent search decision-making for `auto` mode by implementing the [`ConcurrentSearchRequestDecider`](https://github.com/opensearch-project/OpenSearch/blob/c505709d73587fd2fdb714d24510b44ae07aa4a6/server/src/main/java/org/opensearch/search/deciders/ConcurrentSearchRequestDecider.java#L29).
+The search request is parsed using the [`ConcurrentSearchVisitor`](https://github.com/opensearch-project/OpenSearch/blob/c505709d73587fd2fdb714d24510b44ae07aa4a6/server/src/main/java/org/opensearch/search/deciders/ConcurrentSearchVisitor.java) and the [`ConcurrentSearchRequestDecider#evaluateForQuery()`](https://github.com/opensearch-project/OpenSearch/blob/c505709d73587fd2fdb714d24510b44ae07aa4a6/server/src/main/java/org/opensearch/search/deciders/ConcurrentSearchRequestDecider.java#L36) method is invoked from every node of the `QueryBuilder` tree in the search request.
+The final concurrent search decision for the plugin is provided by the [`ConcurrentSearchRequestDecider#getConcurrentSearchDecision()`](https://github.com/opensearch-project/OpenSearch/blob/c505709d73587fd2fdb714d24510b44ae07aa4a6/server/src/main/java/org/opensearch/search/deciders/ConcurrentSearchRequestDecider.java#L44).
+
+If no pluggable deciders are provided, the concurrent search decision is made based on the presence of aggregations in the query with the existing limitations called out in [Limitations](#limitations)   
```

Comment:
The limitations are still respected regardless of the concurrent search deciders right? This sort of implies that deciders can override the existing limitations. We should also clarify that it does not override the `AggregatorFactory` setting from above either.

Reply:
Hmm yes, i see what you mean. I want to capture that the limitations still exist. will not mentioning it be sufficient? Especially since we explicitly say if aggregations are present.. we should qualify that still only supported aggregations will work. Let me see how i can rephrase this

Reply:
Probably we can call that in the begining itself. Tried to frame it [here](https://github.com/opensearch-project/documentation-website/pull/8181/files#r1746094271)